Result/ResultType.swift

```diff
@@ -64,6 +64,30 @@ public extension ResultType {
 	}
 }
 
+/// Protocol used to constrain `tryMap` to `Result`s with compatible `Error`s.
+public protocol ErrorTypeConvertible: ErrorType {
+	typealias ConvertibleType = Self
+	static func errorFromErrorType(error: ErrorType) -> ConvertibleType
+}
+
+public extension ResultType where Error: ErrorTypeConvertible {
+
+	/// Returns the result of applying `transform` to `Success`es’ values, or wrapping thrown errors.
+	public func tryMap<U>(@noescape transform: Value throws -> U) -> Result<U, Error> {
+		return flatMap { value in
+			do {
+				return .Success(try transform(value))
+			}
+			catch {
+				guard let convertedError = Error.errorFromErrorType(error) as? Error else {
```

Comment:
I'm confused about this, if `Error` conforms to `ErrorTypeConvertible`, this is always going to be `non-nil` **and** of type `Error`, right?